File: test-suite/src/test/java/io/micronaut/docs/server/response/ProducesControllerSpec.java
package io.micronaut.docs.server.response;

import io.micronaut.context.ApplicationContext;
import io.micronaut.http.HttpRequest;
import io.micronaut.http.HttpResponse;
import io.micronaut.http.MediaType;
import io.micronaut.http.client.HttpClient;
import io.micronaut.runtime.server.EmbeddedServer;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;

import java.util.Collections;

import static org.junit.Assert.assertEquals;

public class ProducesControllerSpec {

    private static EmbeddedServer server;
    private static HttpClient client;

    @BeforeClass
    public static void setupServer() {
        server = ApplicationContext.run(EmbeddedServer.class, Collections.singletonMap("spec.name", "producesspec"));
        client = server
                .getApplicationContext()
                .createBean(HttpClient.class, server.getURL());
    }

    @AfterClass
    public static void stopServer() {
        if(server != null) {
            server.stop();
        }
        if(client != null) {
            client.stop();
        }
    }

    @Test
    public void testContentTypes() {
        HttpResponse<String> response = client.toBlocking().exchange(HttpRequest.GET("/produces"), String.class);

        assertEquals(MediaType.APPLICATION_JSON_TYPE, response.getContentType().get());

        response = client.toBlocking().exchange(HttpRequest.GET("/produces/html"), String.class);

        assertEquals(MediaType.TEXT_HTML_TYPE, response.getContentType().get());

        response = client.toBlocking().exchange(HttpRequest.GET("/produces/xml"), String.class);

        assertEquals(MediaType.TEXT_XML_TYPE, response.getContentType().get());
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/server/response/ProducesControllerSpec.kt
package io.micronaut.docs.server.response

import io.kotest.matchers.shouldBe
import io.kotest.core.spec.style.StringSpec
import io.micronaut.context.ApplicationContext
import io.micronaut.http.HttpRequest
import io.micronaut.http.MediaType
import io.micronaut.http.client.HttpClient
import io.micronaut.runtime.server.EmbeddedServer

class ProducesControllerSpec: StringSpec() {

    val embeddedServer = autoClose(
            ApplicationContext.run(EmbeddedServer::class.java, mapOf("spec.name" to "producesspec"))
    )

    val client = autoClose(
            embeddedServer.applicationContext.createBean(HttpClient::class.java, embeddedServer.getURL())
    )

    init {
        "test content types"() {
            var response = client.toBlocking().exchange(HttpRequest.GET<Any>("/produces"), String::class.java)

            response.contentType.get() shouldBe MediaType.APPLICATION_JSON_TYPE

            response = client.toBlocking().exchange(HttpRequest.GET<Any>("/produces/html"), String::class.java)

            response.contentType.get() shouldBe MediaType.TEXT_HTML_TYPE

            response = client.toBlocking().exchange(HttpRequest.GET<Any>("/produces/xml"), String::class.java)

            response.contentType.get() shouldBe MediaType.TEXT_XML_TYPE
        }
    }

}
